### PR TITLE
Add --insecure flag to download images command

### PR DESCRIPTION
**Pull request by wongni**
*Issue #, if available:*
#2878 

*Description of changes:*
Add --insecure flag to download images command

*Testing (if applicable):*
Manual test with and without --insecure on `download images` command

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/dependencies/factory.go [hunk 1]
```diff
@@ -579,39 +578,19 @@ func (f *Factory) WithTroubleshoot() *Factory {
...
 
-func (f *Factory) WithHelmSecure() *Factory {
+func (f *Factory) WithHelm(insecure bool) *Factory {
 	f.WithExecutableBuilder()
 
 	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
-		if f.dependencies.HelmSecure != nil {
+		if f.dependencies.Helm != nil {
 			return nil
 		}
```
Comment (abhinavmpandey08) [inline pkg/dependencies/factory.go:587]: The insecure flag here won't work because once the helm executable is built, it is not going to build it again

Comment (vivek-koppuru) [reply pkg/dependencies/factory.go:587]: What do you mean by this? It gets built on every command execution?

Comment (abhinavmpandey08) [reply pkg/dependencies/factory.go]: There was an if condition which would prevent the executable from being re-built if wasn't nil. I think this is resolved in the latest commit

**Review by abhinavmpandey08 — APPROVED** [review]
looks good, thanks!

File: pkg/dependencies/factory.go [hunk 2]
```diff
@@ -596,39 +595,15 @@ func (f *Factory) WithTroubleshoot() *Factory {
 	return f
 }
 
-func (f *Factory) WithHelmSecure() *Factory {
+func (f *Factory) WithHelm(insecure bool) *Factory {
```
Comment (jiayiwang7) [inline pkg/dependencies/factory.go:598]: can we make it optional arg? 
```suggestion
func (f *Factory) WithHelm(opts ...HelmOpts) *Factory {
```

**Review by jiayiwang7 — APPROVED** [review]

**Pull request merged**